packages/platform/atoms/hooks/useUpdateUserTimezone.ts

```diff
@@ -0,0 +1,24 @@
+import { useMutation } from "@tanstack/react-query";
+
+import type { ApiResponse } from "@calcom/platform-types";
+
+type updateTimezoneInput = {
+  timeZone: string;
+  key: string;
+};
+
+export const useUpdateUserTimezone = () => {
+  const mutation = useMutation<ApiResponse<undefined>, unknown, updateTimezoneInput>({
+    mutationFn: (data) => {
+      const { key, timeZone } = data;
+
+      return fetch(`/api/v2/me?apiKey=${key}`, {
```

Comment:
I think we should construct proper query params here to make it expandable if we need 

Check out `new URL(...)`